core/src/data_sources/data_source.rs

```diff
@@ -725,20 +725,42 @@ impl DataSource {
             .await?;
         }
 
+        // Store upsert does not save the text and token count.
+        // These fields don't actually exist in the SQL table.
+        // Because of this, we have to manually construct the UpsertDocument, and save
+        // owned values for text and token count so we can return them.
+        // TODO(@fontanierh): use a different type for "DocumentWithTextAndTokenCount"
+        let doc_text = main_collection_document.text;
+        let doc_token_count = main_collection_document.token_count;
+        let sql_upsert_params = UpsertDocument {
```

Comment:
params maybe. This layer should not know about sql

Reply:
you are right. That's way too many letters for a rust variable :p